test/hotspot/jtreg/runtime/os/TestTimerSlack.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.util.regex.Pattern;
+import java.util.regex.Matcher;
+
+import jdk.test.lib.process.ProcessTools;
+import jdk.test.lib.process.OutputAnalyzer;
+
+/**
+ * @test
+ * @summary Check that timer slack options work
+ * @requires os.family == "linux"
+ * @requires vm.flagless
+ * @library /test/lib
+ * @run driver TestTimerSlack
+ */
+public class TestTimerSlack {
+
+    public static void main(String[] args) throws Exception {
+        int defaultSlack;
+
+        // Check the timer slack value is good by default
+        {
+            ProcessBuilder pb = ProcessTools.createJavaProcessBuilder("-Xlog:os+thread",
+                                                      "TestTimerSlack$Hello");
+
+            OutputAnalyzer output = new OutputAnalyzer(pb.start());
+            output.shouldHaveExitValue(0);
+            output.shouldContain("timer slack:");
+
+            defaultSlack = parseSlackValue(output);
+
+            if (defaultSlack == 0) {
+                output.reportDiagnosticSummary();
+                throw new IllegalStateException("Default slack value (" + defaultSlack + ") is unexpected");
```

Comment:
This pattern is repeated quite a few times. Pity we can't make use of something in OutputAnalyzer directly. A small helper method `fail(String msg)` would avoid the repetition.

Reply:
Done so in the new commit.